src/test/java/org/opensearch/knn/training/VectorReaderTests.java

```diff
@@ -66,59 +64,100 @@ public void testRead_valid_completeIndex() throws InterruptedException, Executio
 
         // Read all vectors and confirm they match vectors
         TestVectorConsumer testVectorConsumer = new TestVectorConsumer();
-        final CountDownLatch inProgressLatch1 = new CountDownLatch(1);
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
         vectorReader.read(
             clusterService,
-            indexName,
-            fieldName,
-            10000,
-            10,
+            DEFAULT_INDEX_NAME,
+            DEFAULT_FIELD_NAME,
+            DEFAULT_MAX_VECTOR_COUNT,
+            DEFAULT_SEARCH_SIZE,
             testVectorConsumer,
-            ActionListener.wrap(response -> inProgressLatch1.countDown(), e -> fail(e.toString()))
+            createOnSearchResponseCountDownListener(inProgressLatch)
         );
 
-        assertTrue(inProgressLatch1.await(100, TimeUnit.SECONDS));
+        assertLatchDecremented(inProgressLatch);
 
         List<Float[]> consumedVectors = testVectorConsumer.getVectorsConsumed();
-        assertEquals(numVectors, consumedVectors.size());
+        assertEquals(DEFAULT_NUM_VECTORS, consumedVectors.size());
 
         List<Float> flatVectors = vectors.stream().flatMap(Arrays::stream).collect(Collectors.toList());
         List<Float> flatConsumedVectors = consumedVectors.stream().flatMap(Arrays::stream).collect(Collectors.toList());
         assertEquals(new HashSet<>(flatVectors), new HashSet<>(flatConsumedVectors));
     }
 
+    public void testRead_valid_trainVectorsIngestedAsIntegers() throws IOException, ExecutionException, InterruptedException {
+        createIndex(DEFAULT_INDEX_NAME);
+        createKnnIndexMapping(DEFAULT_INDEX_NAME, DEFAULT_FIELD_NAME, DEFAULT_DIMENSION);
+
+        // Create list of random vectors and ingest
+        Random random = new Random();
+        List<Integer[]> vectors = new ArrayList<>();
```

Comment:
Is it possible to have mix of lists of different types? If yes I think we need to have such test case with mixed docs.

Reply:
Oh interesting. Hadn't thought of that. I can add this case.

Reply:
Before adding the test case can you please validate do we even allow such kind of vectors as input?

Reply:
Itd be like if someone indexed 2 docs, 1 with [1, 2, 3] and the other with [1.0, 2.0, 3.0]

Reply:
On second thought, I dont think this would need to be added. Given that we process each hit's vector separately, I dont see the need to add a case for when the hits have different representations.